controller/UserController.js

```diff
@@ -0,0 +1,103 @@
+const conn = require("../mariadb");
+const { StatusCodes } = require("http-status-codes");
+const jwt = require("jsonwebtoken");
+const crypto = require("crypto");
+
+const dotenv = require("dotenv");
+
+dotenv.config();
+
+const join = (req, res) => {
+  const { email, password } = req.body;
+  const salt = crypto.randomBytes(10).toString("base64");
+  const hashPassword = crypto
+    .pbkdf2Sync(password, salt, 10000, 10, "sha512")
+    .toString("base64");
+  let sql = "INSERT INTO users (email, password, salt) VALUES (?, ?, ?)";
+  let values = [email, hashPassword, salt];
+  conn.query(sql, values, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+    res.status(StatusCodes.CREATED).json(results);
+  });
+};
+
+const login = (req, res) => {
+  const { email, password } = req.body;
+
+  let sql = "SELECT * FROM users WHERE email = ?";
+
+  conn.query(sql, email, (err, results) => {
+    if (err) {
+      console.log(err);
+      return res.status(StatusCodes.BAD_REQUEST).end();
+    }
+    const loginUser = results[0];
+    const hashPassword = crypto
+      .pbkdf2Sync(password, loginUser.salt, 10000, 10, "sha512")
+      .toString("base64");
+    if (loginUser && loginUser.password == hashPassword) {
+      const token = jwt.sign(
+        {
+          email: loginUser.email,
+        },
+        process.env.PRIVATE_KEY,
+        { expiresIn: "5m", issuer: "sangha" }
+      );
+
+      res.cookie("token", token, {
+        httpOnly: true,
+      });
+      console.log(token);
```

Comment:
```suggestion
      console.log(token);
```
console.log는 코드 올리기 전에 제거 하는 것이 좋습니다.